conversations/impl/src/main/java/org/sakaiproject/conversations/impl/repository/TopicStatusRepositoryImpl.java

```diff
@@ -49,6 +54,66 @@ public Optional<TopicStatus> findByTopicIdAndUserId(String topicId, String userI
 
         return session.createQuery(query).uniqueResultOptional();
     }
+    
+    @Transactional
+    public TopicStatus saveTopicStatus(String topicId, String userId, boolean viewed) {
+        Session session = sessionFactory.getCurrentSession();
+        
+        try {
+            // Try to find the existing record first
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaQuery<TopicStatus> query = cb.createQuery(TopicStatus.class);
+            Root<TopicStatus> root = query.from(TopicStatus.class);
+            query.where(
+                cb.and(
+                    cb.equal(root.get("topic").get("id"), topicId),
+                    cb.equal(root.get("userId"), userId)
+                )
+            );
+            
+            TopicStatus status = session.createQuery(query).uniqueResultOptional().orElse(null);
+            
+            if (status != null) {
+                // Update existing record
+                status.setViewed(viewed);
+                session.merge(status);
+                return status;
```

Comment:
return the managed entity
```suggestion
                return session.merge(status);
```